### PR TITLE
Add better support for RC TX control while in FW AT

### PR DESCRIPTION
**Pull request by chris1seto**
## Overview
Right now, fixed wing auto tune must be triggered by a param edit, and the auto tune process does not allow manual control of the plane. This is unfriendly for a number of reasons:
1. Auto tune is likely to be used during bring up of a new airframe. This is a time when its desirable to always have the RC TX in hand as a quick way to safely deal with control issues. Editing a param means either having a helper or putting the RC TX down and turning attention away from the flying aircraft.
2. The idea of giving up any level of control while autotune is running may be uncomfortable to some users.

## Describe your solution
This PR allows for a selectable aux switch to enable/disable fw_at. It also allows for the user to fly the airplane normally with manual control inputs.

## Test data / coverage
How was it tested? What cases were covered? Logs uploaded to https://review.px4.io/ and screenshots of the important plot parts.

## Additional context
Add any other related context or media.

### CODE REVIEW

File: src/modules/fw_autotune_attitude_control/fw_autotune_attitude_control.hpp [hunk 1]
```diff
@@ -179,6 +179,7 @@ class FwAutotuneAttitudeControl : public ModuleBase<FwAutotuneAttitudeControl>,
 	DEFINE_PARAMETERS(
 		(ParamInt<px4::params::FW_AT_AXES>) _param_fw_at_axes,
 		(ParamBool<px4::params::FW_AT_START>) _param_fw_at_start,
+		(ParamInt<px4::params::FW_AT_RC_AUX_EN>) _param_fw_rc_aux_en,
```
Comment (bresch) [inline src/modules/fw_autotune_attitude_control/fw_autotune_attitude_control.hpp:182]:
```suggestion
		(ParamInt<px4::params::FW_AT_RC_AUX_EN>) _param_fw_at_rc_aux_en,
```

**Review by bresch — CHANGES_REQUESTED** [review]

File: src/modules/fw_autotune_attitude_control/fw_autotune_attitude_control.cpp [hunk 2]
```diff
@@ -237,10 +237,41 @@ void FwAutotuneAttitudeControl::updateStateMachine(hrt_abstime now)
...
+			break;
+		case 5:
+			aux_enable_channel = manual_control_setpoint.aux5;
+			break;
+		case 6:
+			aux_enable_channel = manual_control_setpoint.aux6;
+			break;
+	}
+	
+	bool rc_switch_enabled = aux_enable_channel != NAN &&  aux_enable_channel > .5f;
```
Comment (dagar) [inline src/modules/fw_autotune_attitude_control/fw_autotune_attitude_control.cpp:270]:
```suggestion
	const bool rc_switch_enabled = (aux_enable_channel != NAN) && (aux_enable_channel > .5f);
```

File: src/modules/fw_autotune_attitude_control/fw_autotune_attitude_control.cpp [hunk 3]
```diff
@@ -416,16 +452,12 @@ void FwAutotuneAttitudeControl::updateStateMachine(hrt_abstime now)
...
-	manual_control_setpoint_s manual_control_setpoint{};
-	_manual_control_setpoint_sub.copy(&manual_control_setpoint);
-
 	if (_state != state::wait_for_disarm
 	    && _state != state::idle
-	    && (((now - _state_start_time) > 20_s)
-		|| (fabsf(manual_control_setpoint.x) > 0.2f)
-		|| (fabsf(manual_control_setpoint.y) > 0.2f))) {
+	    && !rc_switch_enabled
+	    && ((now - _state_start_time) > 20_s)) {
```
Comment (bresch) [inline src/modules/fw_autotune_attitude_control/fw_autotune_attitude_control.cpp:460]: I think that if the user doesn't have an RC switch mapped to start and stop auto-tuning, we should still keep the possibility to abort using the sticks

File: src/modules/fw_autotune_attitude_control/fw_autotune_attitude_control.cpp [hunk 4]
```diff
@@ -238,9 +238,45 @@ void FwAutotuneAttitudeControl::updateStateMachine(hrt_abstime now)
...
+	case 5:
+		aux_enable_channel = manual_control_setpoint.aux5;
+		break;
+
+	case 6:
+		aux_enable_channel = manual_control_setpoint.aux6;
+		break;
+	}
+
+	const bool rc_switch_enabled = (aux_enable_channel != NAN) && (aux_enable_channel > .5f);
```
Comment (bresch) [inline src/modules/fw_autotune_attitude_control/fw_autotune_attitude_control.cpp:275]: let's move this in a function that returns a boolean directly

**Review by bresch — CHANGES_REQUESTED** [review]

File: src/modules/fw_autotune_attitude_control/fw_autotune_attitude_control.cpp (hunk 3 above)
Comment (chris1seto) [reply src/modules/fw_autotune_attitude_control/fw_autotune_attitude_control.cpp:460]: What if we aborted on a mode change regardless of if RC aux enable switch is set? I really think it's important for manual control to be possible at all times. I'm open to discussing it, of course. I think anytime we take manual control away from the user it leads to an uncomfortable experience.

File: src/modules/fw_autotune_attitude_control/fw_autotune_attitude_control.cpp (hunk 4 above)
Comment (chris1seto) [reply src/modules/fw_autotune_attitude_control/fw_autotune_attitude_control.cpp:275]: Good point, will do

File: src/modules/fw_autotune_attitude_control/fw_autotune_attitude_control.cpp (hunk 3 above)
Comment (bresch) [reply src/modules/fw_autotune_attitude_control/fw_autotune_attitude_control.cpp]: That sounds like a really good idea actually: aborting on mode change

File: src/modules/fw_autotune_attitude_control/fw_autotune_attitude_control.cpp [hunk 5]
```diff
@@ -230,6 +239,49 @@ void FwAutotuneAttitudeControl::checkFilters()
...
+	case 6:
+		aux_enable_channel = manual_control_setpoint.aux6;
+		break;
+
+	default:
+		aux_enable_channel = NAN;
+		break;
+	}
+
+	return (aux_enable_channel != NAN) && (aux_enable_channel > .5f);
```
Comment (chris1seto) [inline src/modules/fw_autotune_attitude_control/fw_autotune_attitude_control.cpp:282]: The direct comparison of this float to NaN causes a compile time error in the sitl build. Thoughts on how best to refactor to resolve? @dagar

Comment (sfuhrer) [reply src/modules/fw_autotune_attitude_control/fw_autotune_attitude_control.cpp]: You can use PX4_ISFINITE()

**Review by dagar — APPROVED** [review]

**Pull request merged**